src/app/pages/storage2/modules/devices/components/devices/devices.component.ts

```diff
@@ -83,14 +90,24 @@ export class DevicesComponent implements OnInit {
     this.selectedParentItem = undefined;
   }
 
-  onRowSelected(vdev: VDev, event: MouseEvent): void {
-    event.stopPropagation();
-    this.selectedItem = vdev;
-    this.selectedParentItem = [...Object.values(this.topology.data)].find((group: VDev) => {
-      return group?.children.find((child) => {
-        return child.guid === vdev.guid;
-      });
-    });
+  private listenForRouteChanges(id: string): void {
+    if (id && this.dataSource?.data) {
+      const traverseTree = (children: VDev[], parent?: VDev): { item: VDev; parent: VDev } => {
```

Comment:
I noticed that you've inlined the `traverseTree` function into the devices.component. 
We now have another tree traversing function in `app/modules/ix-tree/utils/flatten-tree-with-filter`, which I've used to recursively find multiple items in the tree. It's up to you to merge these 2 methods into a single method